model/trigger/host_spawn.go

```diff
@@ -0,0 +1,194 @@
+	if t.host.StartedBy == evergreen.User {
+		return nil, nil
+	}
+
+	return t.generate(sub)
+}
+
+func (t *spawnHostTriggers) slack() *notification.SlackPayload {
+	text := "Host has failed to spawn"
+	attachment := message.SlackAttachment{
+		Title:     "Click here to spawn another host",
+		TitleLink: spawnHostURL(t.uiConfig.Url),
+		Color:     evergreenFailColor,
+		Fields: []*message.SlackAttachmentField{
+			&message.SlackAttachmentField{
+				Title: "Distro",
+				Value: t.host.Distro.Id,
+				Short: true,
+			},
+		},
+	}
+	if t.host.ProvisionOptions != nil && len(t.host.ProvisionOptions.TaskId) != 0 {
+		attachment.Fields = append(attachment.Fields,
+			&message.SlackAttachmentField{
+				Title: "With data from task",
+				Value: fmt.Sprintf("<%s|%s>", taskLink(&t.uiConfig, t.host.ProvisionOptions.TaskId, -1), t.host.ProvisionOptions.TaskId),
+				Short: true,
+			})
+	}
+
+	if t.host.Provisioned {
+		text = "Host has spawned"
+		attachment.Title = fmt.Sprintf("Evergreen Host: %s", t.host.Id)
+		attachment.TitleLink = spawnHostURL(t.uiConfig.Url)
+		attachment.Color = evergreenSuccessColor
+
+		attachment.Fields = append(attachment.Fields,
+			&message.SlackAttachmentField{
+				Title: "SSH Command",
+				Value: fmt.Sprintf("`%s`", sshCommand(t.host)),
+			})
+	}
+
+	return &notification.SlackPayload{
+		Body:        text,
+		Attachments: []message.SlackAttachment{attachment},
+	}
+}
+
+const spawnHostEmailSubjectTemplate string = `Evergreen Spawn Host with distro '%s' has %s`
+const spawnHostEmailTemplate string = `<html>
+<head>
+</head>
+<body>
+<p>Hi,</p>
+
+<p>The Evergreen Spawn Host <a href="%s">%s</a> with distro '%s' has %s.</p>
+<p>SSH Command: %s</p>
+
+
+</body>
+</html>
+`
+
+func (t *spawnHostTriggers) email() *message.Email {
+	status := "failed to spawn"
+	url := spawnHostURL(t.uiConfig.Url)
+	cmd := "N/A"
+	if t.host.Provisioned {
+		status = "spawned"
+		url = spawnHostURL(t.uiConfig.Url)
+		cmd = sshCommand(t.host)
+	}
+
+	return &message.Email{
+		Subject:           fmt.Sprintf(spawnHostEmailSubjectTemplate, t.host.Distro.Id, status),
+		Body:              fmt.Sprintf(spawnHostEmailTemplate, url, t.host.Id, t.host.Distro.Id, status, cmd),
+		PlainTextContents: false,
+	}
+}
+
+func (t *spawnHostTriggers) makePayload(sub *event.Subscription) interface{} {
+	var payload interface{}
+	if sub.Subscriber.Type == event.SlackSubscriberType {
+		payload = t.slack()
+
+	} else if sub.Subscriber.Type == event.EmailSubscriberType {
+		payload = t.email()
+	}
+
+	return payload
+}
+
+func (t *spawnHostTriggers) generate(sub *event.Subscription) (*notification.Notification, error) {
+	payload := t.makePayload(sub)
+	grip.Info(message.Fields{
+		"look-here":   "foidsjfhkafhaiwf",
+		"pl":          payload == nil,
+		"e":           t.event.EventType,
+		"provisioned": t.host.Provisioned,
+	})
+	if payload == nil {
+		return nil, nil
```

Comment:
returning nil, nil has been the source of a bunch of our bugs, this is probably unavoidable, but if its easy enough to make this case return an error, we probably should.